### ISSUE TITLE
[xz / liblzma] CMake cannot find LibLZMA

### ISSUE THREAD

**Issue by wolfv** [microsoft/vcpkg#22807]
The hard-coded CMake uses hte `${_IMPORT_PREFIX}` variable which is cleaned up in the generated CMake and thus not anymore accessible. That results in CMake being unable to find LibLZMA

https://github.com/microsoft/vcpkg/blob/c76e5f092a3919edd52ea81e6c7a75af5cff7e39/ports/liblzma/portfile.cmake#L48

This variable is cleaned up in line 77 of `liblzmaTargets.cmake`

```
# Cleanup temporary variables.
set(_IMPORT_PREFIX)
```

Not sure how to best fix this.

**Comment by JackBoosY** [comment]
Yeah `_IMPORT_PREFIX` is cleaned, but see the code:
```cmake
# Compute the installation prefix relative to this file.
get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
if(_IMPORT_PREFIX STREQUAL "/")
  set(_IMPORT_PREFIX "")
endif()

# Create imported target liblzma::liblzma
add_library(liblzma::liblzma SHARED IMPORTED)

set_target_properties(liblzma::liblzma PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "Threads::Threads"
)

if(CMAKE_VERSION VERSION_LESS 2.8.12)
  message(FATAL_ERROR "This file relies on consumers using CMake 2.8.12 or greater.")
endif()

# Load information for each installed configuration.
get_filename_component(_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
file(GLOB CONFIG_FILES "${_DIR}/liblzmaTargets-*.cmake")
foreach(f ${CONFIG_FILES})
  include(${f})
endforeach()

# Cleanup temporary variables.
set(_IMPORT_PREFIX)
```
It's cleaned after all the usage. I think it doesn't have any problems, can you please provide the failure logs?

**Comment by dg0yt** [comment]
The problem is that the portfile appends references to `${_IMPORT_PREFIX}` to `liblzmaConfig.cmake`. And it wouldn't  even work with *appending*  to `liblzmaTargets.cmake` due to the generated `unset`.

The needed fix is to append *absolute paths*. These would be recognized by  `vcpkg_cmake_config_fixup` (when called after appending), and that script would replace the absolute part with `VCPKG_IMPORT_PREFIX`, a variable which is set (and not unset) in each script that uses it.

**Comment by wolfv** [comment]
Thanks for getting back so promptly. Yes, the problem is that the `${_IMPORT_PREFIX}` is referenced in the `liblzmaConfig.cmake` file. 

The error log is just that `find_package(LibLZMA REQUIRED)` fails because it can't find `lzma.h` (since the `PATHS` resolves to `/include` which doesn't exist).

**Comment by JackBoosY** [comment]
This issue should be fixed in #22856.
